integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -124,40 +133,62 @@ impl TestReshardingParametersBuilder {
             .cloned()
             .collect();
 
-        let non_validator_accounts: Vec<_> = accounts
-            .iter()
-            .filter(|account| !block_and_chunk_producers.contains(account))
-            .collect();
-        assert!(non_validator_accounts.len() >= 2);
-        let archival_clients = vec![non_validator_accounts[0].clone()];
-        let rpc_clients = vec![non_validator_accounts[1].clone()];
-        let clients =
-            vec![block_and_chunk_producers.clone(), archival_clients.clone(), rpc_clients.clone()]
-                .into_iter()
-                .flatten()
-                .collect();
+        // Split the clients into producers, validators, rpc and archivals node.
+        let tmp = clients.clone();
+        let (producers, tmp) = tmp.split_at(num_producers as usize);
+        let producers = producers.to_vec();
+        let (validators, tmp) = if num_validators > 0 {
+            let (validators, tmp) = tmp.split_at(num_validators as usize);
+            (validators.to_vec(), tmp)
+        } else {
+            (vec![], tmp)
+        };
```

Comment:
Does not `let (validators, tmp) = tmp.split_at(num_validators as usize);` just work?

Reply:
Yeah that works!